### PR TITLE
Get the version from package.json only

### PR DESCRIPTION
**Pull request by babich-a**

### CODE REVIEW

File: build/gulp/context.js [hunk 1]
```diff
@@ -1,9 +1,9 @@
-var fs = require('fs');
 var argv = require('yargs')
     .default('uglify', false)
     .argv;
 
-var productVersion = readVersion(),
+var productVersion = require('../../package.json').version,
+    scriptVersion = productVersion,
```
Comment (nightskylark) [inline build/gulp/context.js:6]: Let's get rid of `scriptVersion`  variable at all.

File: build/gulp/js-bundles.js [hunk 2]
```diff
@@ -46,7 +49,13 @@ function processDevBundles(bundles) {
 function muteWebPack() {
 }
 
-gulp.task('js-bundles-prod', ['transpile'], function() {
+gulp.task('version-fix', ['transpile'], function() {
+    return gulp.src(context.TRANSPILED_PATH + '/' + versionPath, { base: './' })
```
Comment (nightskylark) [inline build/gulp/js-bundles.js]: `path.join`?

File: build/gulp/context.js (hunk 1 above)
Comment (babich-a) [reply build/gulp/context.js:6]: We need unchanged (scriptVersion) version from product.json, but productVersion will be changed.

**Review by nightskylark — DISMISSED** [review]

File: build/gulp/js-bundles.js [hunk 3]
```diff
@@ -31,6 +32,8 @@ var DEBUG_BUNDLES = BUNDLES.concat([
     '/bundles/dx.custom.js'
 ]);
 
+var versionPath = 'core/version.js';
```
Comment (AlekseyMartynov) [inline build/gulp/js-bundles.js]: Constant? `VERSION_FILE_PATH`

File: js/core/version.js [hunk 4]
```diff
@@ -1 +1 @@
-module.exports = "19.1.0"; /* DevExpress.VERSION */
+module.exports = "19.1"; /* DevExpress.VERSION */
```
Comment (AlekseyMartynov) [inline js/core/version.js]: - Is the marker comment still required?
- Should we use placeholder `0.0.0` instead so we never need to patch it manually?

Comment (babich-a) [reply js/core/version.js]: Placeholder 0.0.0 is a good idea.

File: build/gulp/context.js (hunk 1 above)
Comment (AlekseyMartynov) [reply build/gulp/context.js:6]: Do we need all three?
`productVersion`, `scriptVersion`, `packageVersion`

File: build/gulp/js-bundles.js [hunk 5]
```diff
@@ -46,7 +49,13 @@ function processDevBundles(bundles) {
 function muteWebPack() {
 }
 
-gulp.task('js-bundles-prod', ['transpile'], function() {
+gulp.task('version-fix', ['transpile'], function() {
```
Comment (San4es) [inline build/gulp/js-bundles.js]: Maybe 'version-fix' -> 'version-replace'/'replace-version'/'version-replacement'/'version-substitution'/'substitute-version'?

File: build/gulp/js-bundles.js [hunk 6]
```diff
@@ -46,7 +49,13 @@ function processDevBundles(bundles) {
...
+    return gulp.src(path.join(context.TRANSPILED_PATH, VERSION_FILE_PATH), { base: './' })
+        .pipe(replace(/"\d\d\.\d.*?"/, `"${context.version.script}"`))
```
Comment (San4es) [inline build/gulp/js-bundles.js]: Will this regex /"\d\d\.\d.*?"/ work for the "0.0.0" version placeholder? 

I see this in transpiled scripts and bundles:

    module.exports = "0.0.0";

Comment (AlekseyMartynov) [reply build/gulp/js-bundles.js]: `/[\d.]+/` this should work

Comment (AlekseyMartynov) [reply build/gulp/js-bundles.js]: `/"[\d.]+"/` this should work

File: js/core/version.js [hunk 7]
```diff
@@ -1 +1 @@
-module.exports = "19.1.0"; /* DevExpress.VERSION */
+module.exports = "0.0.0";
```
Comment (San4es) [inline js/core/version.js]: Can we use a more specific placeholder (smth like "%VERSION%", "#VERSION_TO_REPLACE#", "{VERSION_PLACEHOLDER}", "$VERSION_DUMMY$", etc) to make this magic constant more clear and avoid unwanted potential replacements? 🤔

File: build/gulp/context.js (hunk 1 above)
Comment (San4es) [reply build/gulp/context.js:6]: Taking into account this:
https://github.com/DevExpress/DevExtreme/blob/941bc99c0fccfbb9ca9720e54d880c62f841cb62/build/gulp/context.js#L46-L48

looks like we can kill `scriptVersion` and  use `packageVersion` instead now.

File: js/core/version.js (hunk 7 above)
Comment (babich-a) [reply js/core/version.js]: I like the pattern. We need no non-searchable regular expression/

File: build/gulp/context.js (hunk 1 above)
Comment (babich-a) [reply build/gulp/context.js:6]: > 
> 
> Taking into account this:
> 
> [DevExtreme/build/gulp/context.js](https://github.com/DevExpress/DevExtreme/blob/941bc99c0fccfbb9ca9720e54d880c62f841cb62/build/gulp/context.js#L46-L48)
> 
> Lines 46 to 48 in [941bc99](/DevExpress/DevExtreme/commit/941bc99c0fccfbb9ca9720e54d880c62f841cb62)
>  if(packageVersion !== scriptVersion) { 
>      throw 'Product version mismatch in ' + SCRIPT_VERSION_FILE; 
>  } 
> 
> looks like we can kill `scriptVersion` and use `packageVersion` instead now.

Your example is just reading from package.json and validation. Look at https://github.com/DevExpress/DevExtreme/blob/941bc99c0fccfbb9ca9720e54d880c62f841cb62/build/gulp/context.js#L23 and https://github.com/DevExpress/DevExtreme/blob/941bc99c0fccfbb9ca9720e54d880c62f841cb62/build/gulp/context.js#L24.

Hotfix, for example, will have:

package version: `18.2.4-pre-10000`
product version: `18.2.3 (build 10000)`
script version: `18.2.3` <--DevExpress.VERSION

**Review by San4es — APPROVED** [review]
LGTM

**Review by BingoRUS — APPROVED** [review]

**Pull request merged**